pkg/healthz/healthz.go

```diff
@@ -0,0 +1,47 @@
+package healthz
+
+import (
+	"net/http"
+	"time"
+
+	"golang.org/x/net/context"
+	"k8s.io/klog"
+)
+
+type HealthzServer struct {
+	httpServer *http.Server
+}
+
+func New(address string) *HealthzServer {
+	healthServer := &HealthzServer{
+		httpServer: &http.Server{Addr: address},
+	}
+	healthServer.httpServer.Handler = healthServer
+
+	return healthServer
+}
+
+func (hs *HealthzServer) Run(stop <-chan struct{}) {
+	go func() {
+		err := hs.httpServer.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			klog.Errorf("Unable to start healthz server: %s", err.Error())
```

Comment:
It seems it should return here and avoid trying to call `Shutdown` below if the httpServer never started.

Reply:
I don't understand what you mean :)

Reply:
Sorry I missed that it's a go-routine - I originally misread it as simply calling an anonymous function - still getting used to reading Go code plus it's Fri  😄

Reply:
hehe np